es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java

```diff
@@ -627,4 +630,73 @@ private void addSortOptionToSearchRequest(SearchRequestBuilder searchRequestBuil
             .collect(Collectors.toCollection(LinkedList::new));
     }
 
+    @Override
+    public List<Message> getMessages(String queue) {
+        try {
+            Expression expression = Expression.fromString("queue='" + queue + "'");
+            QueryBuilder queryBuilder = expression.getFilterBuilder();
+
+            BoolQueryBuilder filterQuery = QueryBuilders.boolQuery().must(queryBuilder);
+            QueryStringQueryBuilder stringQuery = QueryBuilders.queryStringQuery("*");
+            BoolQueryBuilder fq = QueryBuilders.boolQuery().must(stringQuery).must(filterQuery);
+
+            final SearchRequestBuilder srb = elasticSearchClient.prepareSearch(logIndexPrefix + "*")
+                    .setQuery(fq)
+                    .setTypes(MSG_DOC_TYPE)
+                    .addSort(SortBuilders.fieldSort("created").order(SortOrder.ASC));
+
+            return mapGetMessagesResponse(srb.execute().actionGet());
+        } catch (Exception e) {
+            logger.error("Failed to get messages for queue: {}", queue, e);
+        }
+        return null;
+    }
+
+    private List<Message> mapGetMessagesResponse(SearchResponse response) throws IOException {
+        SearchHit[] hits = response.getHits().getHits();
+        TypeFactory factory = TypeFactory.defaultInstance();
+        MapType type = factory.constructMapType(HashMap.class, String.class, String.class);
+        List<Message> messages = new ArrayList<>(hits.length);
+        for (SearchHit hit : hits) {
+            String source = hit.getSourceAsString();
+            Map<String, String> mapSource = objectMapper.readValue(source, type);
+            Message msg = new Message(mapSource.get("messageId"), mapSource.get("payload"), null);
+            messages.add(msg);
+        }
+        return messages;
+    }
+
+    @Override
+    public List<EventExecution> getEventExecutions(String event) {
+        try {
+            Expression expression = Expression.fromString("event='" + event + "'");
+            QueryBuilder queryBuilder = expression.getFilterBuilder();
+
+            BoolQueryBuilder filterQuery = QueryBuilders.boolQuery().must(queryBuilder);
+            QueryStringQueryBuilder stringQuery = QueryBuilders.queryStringQuery("*");
+            BoolQueryBuilder fq = QueryBuilders.boolQuery().must(stringQuery).must(filterQuery);
+
+            final SearchRequestBuilder srb = elasticSearchClient.prepareSearch(logIndexPrefix + "*")
+                    .setQuery(fq).setTypes(EVENT_DOC_TYPE)
+                    .addSort(SortBuilders.fieldSort("created")
+                            .order(SortOrder.ASC));
+
+            return mapEventExecutionsResponse(srb.execute().actionGet());
+        } catch (Exception e) {
+            logger.error("Failed to get executions for event: {}", event, e);
+        }
+        return null;
```

Comment:
Please throw an exception instead of returning null.

Reply:
Done.